### PR TITLE
Rename Vmax and Concentration

### PR DESCRIPTION
**Pull request by robjmcgibbon**
When calculating the unsoftened vmax we weren't skipping the first particle (the one at the origin). I think this is due to differences in the value of the halo centre vs the most bound particle position because of compression. For some reason this is only a problem for HBT and SubFind, not VR.

When calculating unsoftened vmax we now always skip the first particle.

I've also renamed `MaximumCircularVelocitySoft` to `MaximumCircularVelocity`, as this is the default people should use. I've done the same for concentration.

### CODE REVIEW

File: parameter_files/COLIBRE.yml [hunk 1]
```diff
@@ -321,6 +325,7 @@ SOProperties:
     KineticEnergyGas: true
     KineticEnergyStars: true
     MassFractionSatellites: true
+    MassFractionExternal: true
```
Comment (VictorForouhar) [inline parameter_files/COLIBRE.yml:328]: What is this property relevant to the PR?

Comment (robjmcgibbon) [reply parameter_files/COLIBRE.yml:328]: It was just missing from the COLIBRE parameter file, so I added it in. It was being calculated previously for COLBIRE

File: property_table.py [hunk 2]
```diff
@@ -2499,12 +2499,12 @@ class PropertyTable:
...
         ),
-        "concentration": (
-            "Concentration",
+        "concentration_unsoft": (
+            "ConcentrationUnsoftened",
             1,
             np.float32,
             "dimensionless",
-            "Halo concentration assuming an NFW profile.",
+            "Halo concentration assuming an NFW profile. No particle softening.",
```
Comment (VictorForouhar) [inline property_table.py:2507]: The softening explanations in the property tables are somewhat inconsistent in their presentation, e.g. `No particle softening`, `not accounting for softening length.`. Perhaps harmonise?

File: SO_properties.py [hunk 3]
```diff
@@ -2522,9 +2523,9 @@ class SOProperties(HaloProperty):
             "starOfrac",
             "starFefrac",
             "gasmetalfrac_SF",
-            "concentration",
+            "concentration_unsoft",
             "concentration_soft",
```
Comment (VictorForouhar) [inline SO_properties.py:2527]: Does this not need changing from `concentration_soft` to `concentration`?

Comment (robjmcgibbon) [reply SO_properties.py:2527]: I'm keeping everything explicit, so the softened concentration function is still called concentration_soft

File: subhalo_properties.py [hunk 4]
```diff
@@ -932,31 +934,34 @@ def veldisp_matrix_dm(self) -> unyt.unyt_array:
...
+                self.mass_dm, soft_r
             )
-        return self.DM_vmax
+        return self.DM_vmax_soft
 
     @lazy_property
-    def DM_R_vmax(self) -> unyt.unyt_quantity:
+    def DM_R_vmax_soft(self) -> unyt.unyt_quantity:
         """
         Radius for which the maximum circular velocity of dark matter particles
```
Comment (VictorForouhar) [inline subhalo_properties.py:954]: Perhaps update comment on function to say that it places all particles within softening at a radius equal to the softening.

Comment (VictorForouhar) [reply subhalo_properties.py:954]: Or say this is Rmax for softened Vmax

File: subhalo_properties.py [hunk 5]
```diff
@@ -593,30 +593,32 @@ def vcom(self) -> unyt.unyt_array:
...
+        if not hasattr(self, "r_vmax_unsoft"):
+            self.r_vmax_unsoft, self.vmax_unsoft = get_vmax(self.mass, self.radius, nskip=1)
+        return self.r_vmax_unsoft
 
     @lazy_property
-    def Vmax(self) -> unyt.unyt_quantity:
+    def Vmax_unsoft(self) -> unyt.unyt_quantity:
         """
         Maximum circular velocity of the halo.
+        No softening is applied to the particle radii.
```
Comment (VictorForouhar) [inline subhalo_properties.py:613]: I would say that all particles retain their original positions, and are not constrained to be at least a softening distance away. Otherwise someone may interpret this as smoothing the particle out using some sort of a kernel?

File: parameter_files/COLIBRE.yml (hunk 1 above)
Comment (VictorForouhar) [reply parameter_files/COLIBRE.yml:328]: COLIBRE*

File: subhalo_properties.py (hunk 5 above)
Comment (robjmcgibbon) [reply subhalo_properties.py]: [344dd9a](https://github.com/SWIFTSIM/SOAP/pull/93/commits/344dd9a053a1c5de4ecd50a90f308c5dd1de7aca)

File: property_table.py (hunk 2 above)
Comment (robjmcgibbon) [reply property_table.py:2491]: [344dd9a](https://github.com/SWIFTSIM/SOAP/pull/93/commits/344dd9a053a1c5de4ecd50a90f308c5dd1de7aca)

**Comment by VictorForouhar** [comment]
Looks good to me.
